bap-vibes/src/linear_ssa.ml

```diff
@@ -67,77 +67,144 @@ let congruent (a : var) (b : var) : bool =
   if String.is_empty name_1 && String.is_empty name_2 then false
   else String.equal name_1 name_2
 
-let rec linearize_exp ~(prefix : string) (exp : Bil.exp) : Bil.exp =
-  match exp with
+module Linear = struct
+
+  module Env = struct
+
+    type t = {
+      vars : Var.Set.t;
+      prefix : string;
+    }
+
+    let add_var v env = {env with vars = Set.add env.vars v}
+    let with_prefix prefix env = {env with prefix}
+
+  end
+
+  include Monad.State.T1(Env)(Knowledge)
+  include Monad.State.Make(Env)(Knowledge)
```

Comment:
It's not clear to me that you use the Knowledge base monad in the `linearize` code? Do you need it?